crates/router/src/core/payment_methods.rs

```diff
@@ -1767,20 +1767,16 @@ pub async fn update_payment_method(
         },
     )?;
 
-    let pmd: pm_types::PaymentMethodVaultingData = cards::retrieve_payment_method_from_vault(
-        &state,
-        &merchant_account,
-        &payment_method.customer_id,
-        &payment_method,
-    )
-    .await
-    .change_context(errors::ApiErrorResponse::InternalServerError)
-    .attach_printable("Failed to retrieve payment method from vault")?
-    .data
-    .expose()
-    .parse_struct("PaymentMethodCreateData")
-    .change_context(errors::ApiErrorResponse::InternalServerError)
-    .attach_printable("Failed to parse PaymentMethodCreateData")?;
+    let pmd: pm_types::PaymentMethodVaultingData =
+        cards::retrieve_payment_method_from_vault(&state, &merchant_account, &payment_method)
```

Comment:
Payment_method is not inside vault, only the payment_method_data is inside the vault so change the function name, and these functions should not in cards module. Because one of the payment_method data is card, not all payment_method_data is card